### DISCUSSION TITLE
How to debug?

### DISCUSSION THREAD

**Discussion by kirushyk** [mesonbuild/meson#9639] [General]
Is there a way to set a breakpoint and see what some particular list contains?

**Comment by eli-schwartz** [comment]
I am a personal fan of printf-style debugging.

```meson
mylist = ['foo', 'bar']
message('mylist contains: @0@'.format(mylist))
```

Meson doesn't include an interactive meson.build debugger. If you want to debug meson itself I guess you can run meson with pdb.

**Reply by steevenlee** [reply]
It doesn't break when some error happens, also the command 'import pdb' and 'breakpoint()' failed.
meson disabled standard python API?
